File: library/src/main/java/com/android/orient/practice/kldf/kldf/BaseApplication.java
package com.android.orient.practice.kldf.kldf;

import android.app.Application;

import com.android.orient.practice.kldf.kldf.util.CacheUtil;

/**
 * Created by zhanglei on 2017/4/17.
 */

public class BaseApplication extends Application {

    public static Application instance;

    public static Application getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        CacheUtil.initCache(this);
    }
}

File: app/src/main/java/com/android/orient/sports/happy/application/BaseApplication.kt
package com.android.orient.sports.happy.application

import android.app.Application
import com.android.orient.sports.happy.utils.ContextDelegate
import com.android.orient.sports.happy.utils.startAlarmService

/**
 * Package Name: com.android.orient.sports.happysports.application
 * Date: 2018/11/5 16:08
 * Author: zhanglei
 */
class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        ContextDelegate.context = this
        startAlarmService(this)
    }

}